recipe/run_test.sh

```diff
@@ -1,48 +1,45 @@
 #!/bin/bash
-set -x
-
-command -v mpichversion
-mpichversion
-
-command -v mpicc
-mpicc -show
-
-command -v mpicxx
-mpicxx -show
-
-command -v mpif90
-mpif90 -show
-
-command -v mpiexec
+set -ex
 
+export HYDRA_LAUNCHER=fork
+MPIEXEC="${PWD}/mpiexec.sh"
 
 pushd "tests"
 
-function mpi_exec() {
-  # use pipes to avoid O_NONBLOCK issues on stdin, stdout
-  mpiexec -launcher fork $@ 2>&1 </dev/null | cat
-}
-
 if [[ $PKG_NAME == "mpich" ]]; then
-  mpi_exec -n 4 python test_exec.py
+  command -v mpichversion
+  mpichversion
+
+  command -v mpiexec
+  $MPIEXEC -n 1 mpivars
+  $MPIEXEC -n 4 ./helloworld.sh
 fi
 
 if [[ $PKG_NAME == "mpich-mpicc" ]]; then
+  command -v mpicc
+  mpicc -show
+
   mpicc $CFLAGS $LDFLAGS helloworld.c -o helloworld_c
-  mpi_exec -n 4 ./helloworld_c
+  $MPIEXEC -n 4 ./helloworld_c
 fi
 
 if [[ $PKG_NAME == "mpich-mpicxx" ]]; then
+  command -v mpicxx
+  mpicxx -show
+
   mpicxx $CXXFLAGS $LDFLAGS helloworld.cxx -o helloworld_cxx
-  mpi_exec -n 4 ./helloworld_cxx
+  $MPIEXEC -n 4 ./helloworld_cxx
 fi
 
 if [[ $PKG_NAME == "mpich-mpifort" ]]; then
-  mpif77 $FFLAGS $LDFLAGS helloworld.f -o helloworld_f
-  mpi_exec -n 4 ./helloworld_f
+  command -v mpifort
+  mpifort -show
+
+  mpifort $FFLAGS $LDFLAGS helloworld.f -o helloworld_f
+  $MPIEXEC -n 4 ./helloworld_f
 
-  mpif90 $FFLAGS $LDFLAGS helloworld.f90 -o helloworld_f90
```

Comment:
Is there a reason you decided to delete these tests?

Reply:
Both MPICH and Open MPI converged to provide `mpifort` as the Fortran compiler wrapper. In MPICH 3.2.1, both `mpif77` and `mpif90` are just symbolic links to `mpifort`. 

Please note I did not remove any tests, I'm just using the `mpifort` wrapper for both Fortran examples.

Reply:
Previously, these symlinks were tested to exist and work, now they are no longer tested. Can you put these back?